crates/ibc/src/lib.rs

```diff
@@ -695,24 +740,91 @@ where
                     .map_err(|e| Error::Handler(Box::new(e)))?;
 
                 // Extract MASP tx from the memo in the packet if needed
-                let masp_tx = match &*envelope {
+                let (masp_tx, tokens) = match &*envelope {
                     MsgEnvelope::Packet(PacketMsg::Recv(msg))
                         if self
                             .is_receiving_success(msg)?
                             .is_some_and(|ack_succ| ack_succ) =>
                     {
-                        extract_masp_tx_from_packet(&msg.packet)
+                        let tokens = if msg.packet.port_id_on_b.as_str()
```

Comment:
I know the `execute` function is already big. Can we add a helper function to retrieve the tokens like `extract_masp_tx_from_packet()`?
It would be better to make the helper to get the received `vec![ibc_denom]` or `ibc_traces` from the packet.

Reply:
Done thanks. Made a separate function `msg_recv_packet_tokens` to extract the tokens used in an IBC receive packet message.